test/common/network_policy_access_ns_to_svc.go

```diff
@@ -0,0 +1,278 @@
+		ObjectMeta: v1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: projectName,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort{
+					Protocol:   corev1.ProtocolTCP,
+					Port:       80,
+					TargetPort: intstr.FromInt(8080),
+				},
+			},
+			Selector: getTestingCurlLabels(),
+		},
+	}
+
+	// check if service exist
+	key := k8sclient.ObjectKey{Name: serviceCR.GetName(), Namespace: serviceCR.GetNamespace()}
+	err := ctx.Client.Get(goctx.TODO(), key, serviceCR)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return nil, fmt.Errorf("error occured while retrieving service %s : %v", serviceCR.GetName(), err)
+	} else if k8serr.IsNotFound(err) {
+		err = resources.DoOpenshiftCreateServiceInANamespace(openshiftAPIURL, token, projectName, serviceCR)
+		if err != nil {
+			return nil, fmt.Errorf("error occured while making request to create service %s : %v", serviceCR.GetName(), err)
+		}
+	}
+
+	return serviceCR, nil
+}
+
+func createPodWithAnEndpoint(ctx *TestingContext, openshiftAPIURL string, token string) (*corev1.Pod, error) {
+
+	podCR := &corev1.Pod{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      podName,
+			Labels:    getTestingCurlLabels(),
+			Namespace: projectName,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				corev1.Container{
+					Name:  containerName,
+					Image: "busybox:1.31.1",
+					Command: []string{
+						"/bin/sh",
+					},
+					Args: []string{
+						"-c",
+						fmt.Sprintf("while true ; do  echo -e \"HTTP/1.1 200 OK\n\n %s \" | nc -l -p 8080  ; done", podEndpoitResponse),
+					},
+					Ports: []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 8080,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// check if pod exist
+	key := k8sclient.ObjectKey{Name: podCR.GetName(), Namespace: podCR.GetNamespace()}
+	err := ctx.Client.Get(goctx.TODO(), key, podCR)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return nil, fmt.Errorf("error occured while retrieving pod %s : %v", podCR.GetName(), err)
+	} else if k8serr.IsNotFound(err) {
+		err = resources.DoOpenshiftCreatePodInANamespace(openshiftAPIURL, token, projectName, podCR)
+		if err != nil {
+			return nil, fmt.Errorf("error occured while making request to create pod %s : %v", podCR.GetName(), err)
+			// t.Fatalf("error occured while creating pod: %v", err)
+		}
+	}
+
+	return podCR, nil
+
+}
+
+func checkPodStatus(ctx *TestingContext, podCR *corev1.Pod) (bool, error) {
+	checkPodStatus := false
+	key := k8sclient.ObjectKey{Name: podCR.GetName(), Namespace: podCR.GetNamespace()}
+	for !checkPodStatus {
+		err := ctx.Client.Get(goctx.TODO(), key, podCR)
+		if err != nil {
+			return false, fmt.Errorf("error getting pod: %v", err)
+		}
+		for _, cnd := range podCR.Status.Conditions {
+			if cnd.Type == corev1.ContainersReady {
+				if cnd.Status != corev1.ConditionStatus("True") {
+					checkPodStatus = false
+				} else {
+					checkPodStatus = true
+					break
+				}
```

Comment:
Here is a similar thing. Would it not be better to check for the passing statement first and then handle a failing statement.
```
 if cnd.Status == corev1.ConditionStatus("True") {
					checkPodStatus = true
					break
				} else {
					checkPodStatus = false
				}
```